### PR TITLE
[4972] Improve Search behavior in Browse Components tool of Page Builder

### PR DESCRIPTION
**Pull request by aachavarria**
BrowseComponentsPanel: Moving styles and translation to his own file.
Updates on BrowseComponents Panel to use all contentTypes filter.
fetchItemsByContentType service: fixing service to return only components of the selected contentType;

https://github.com/craftercms/craftercms/issues/4972

### CODE REVIEW

File: ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx [hunk 1]
```diff
@@ -179,7 +87,10 @@ export default function PreviewBrowseComponentsPanel() {
         const items =
           source.page[source.pageNumber]
             ?.map((id: string) => source.byId[id])
-            .filter((item: ContentInstance) => item.craftercms.contentTypeId === source.contentTypeFilter) || [];
+            .filter(
+              (item: ContentInstance) =>
+                source.contentTypeFilter === 'all' || item.craftercms.contentTypeId === source.contentTypeFilter
```
Comment (rart) [inline ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx]: - too inefficient to iterate through the whole array when you know in advanced that you needn't apply the filter at all
- also, the `||` is misplaced; it doesn't avoid an exception if the .map returns undefined
```js
let items = source.page[source.pageNumber]?.map(...) ?? []
if (source.contenTypeFilter !== 'all') 
  items = items.filter(...)

File: ui/app/src/state/epics/contentTypes.ts [hunk 2]
```diff
@@ -28,29 +28,30 @@ import { fetchItemsByContentType } from '../../services/content';
...
-        state.sites.active,
-        state.preview.components.contentTypeFilter,
-        state.contentTypes.byId,
-        state.preview.components.query
-      ).pipe(map(fetchComponentsByContentTypeComplete), catchAjaxError(fetchComponentsByContentTypeFailed))
-    )
-  );
-
-export default [fetch, fetchComponentsByContentType] as Epic[];
+] as Epic[];
```
Comment (rart) [inline ui/app/src/state/epics/contentTypes.ts]:
```suggestion
] as CrafterCMSEpic[];
```

File: ui/app/src/state/actions/preview.ts [hunk 3]
```diff
@@ -247,15 +247,9 @@ export const fetchAssetsPanelItemsComplete = /*#__PURE__*/ createAction<SearchRe
...
-  contentTypeFilter?: string,
-  options?: Partial<ComponentsContentTypeParams>
-): StandardAction {
-  return {
-    type: FETCH_COMPONENTS_BY_CONTENT_TYPE,
-    payload: { contentTypeFilter, options }
-  };
-}
+export const fetchComponentsByContentType = /*#__PURE__*/ createAction<Partial<ComponentsContentTypeParams>>(
+  FETCH_COMPONENTS_BY_CONTENT_TYPE
```
Comment (rart) [inline ui/app/src/state/actions/preview.ts]: we could/should discard these constants

File: ui/app/src/services/content.ts [hunk 4]
```diff
@@ -462,31 +462,37 @@ export function fetchItemsByContentType(
...
   }
+
   return postJSON(`/studio/api/2/search/search.json?siteId=${site}`, {
     ...reversePluckProps(options, 'type'),
     filters: { 'content-type': contentTypes }
   }).pipe(
     map<AjaxResponse, { count: number; paths: string[] }>(({ response }) => ({
       count: response.result.total,
-      paths: response.result.items.filter((item) => item.type === options.type).map((item) => item.path)
+      paths: response.result.items.map((item) => item.path)
```
Comment (rart) [inline ui/app/src/services/content.ts:472]: Looks like you're dropping support for the type filter. why?
if we do it, then you need to drop it from the options of this method too.

File: ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx [hunk 5]
```diff
@@ -296,60 +201,46 @@ interface BrowsePanelUIProps {
 }
 
 function BrowsePanelUI(props: BrowsePanelUIProps) {
-  const { componentsResource, classes, onPageChanged, onDragStart, onDragEnd, chooseContentTypeImageUrl } = props;
+  const { componentsResource, classes, onPageChanged, onDragStart, onDragEnd } = props;
```
Comment (rart) [inline ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx]: Let's break the individual components into files

File: ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx [hunk 6]
```diff
@@ -296,60 +201,46 @@ interface BrowsePanelUIProps {
...
+            onDragStart={() => onDragStart(item)}
+            onDragEnd={onDragEnd}
+          />
+        ))}
+      </List>
+      {count === 0 && (
+        <EmptyState
+          title={formatMessage(translations.noResults)}
+          classes={{ image: classes.noResultsImage, title: classes.noResultsTitle }}
         />
```
Comment (rart) [inline ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx]: When there aren't any results, is there an empty message elsewhere? 
I'd also like to add a note, that only shared components are shown here. Let's discuss.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/services/content.ts (hunk 4 above)
Comment (aachavarria) [reply ui/app/src/services/content.ts:472]: I was doing some testing and the result from the fetch is only including components, so looks like the filter is not doing anything, i think in the past the fetch result included groovy or something similar.

File: ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx (hunk 6 above)
Comment (aachavarria) [reply ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx]: there is an  `*<EmptyState/>`, let's discuss tomorrow

**Review by rart — APPROVED** [review]

**Pull request merged**